bpf-gpl/tc.c

```diff
@@ -535,11 +537,28 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 		state.flags |= CALI_ST_NAT_OUTGOING;
 	}
 
+	/* We are possibly past (D)NAT, but that is ok, we need to let the IP
+	 * stack do the RPF check on the source, dest is not importatnt.
+	 */
+	if (CALI_F_TO_HOST && ct_result_rpf_failed(state.ct_result.rc)) {
+		fwd_fib_set(&fwd, false);
+	}
+
+
 	/* skip policy if we get conntrack hit */
 	if (state.ct_result.rc != CALI_CT_NEW) {
```

Comment:
Does this check need to mask the rc too?

Reply:
technically no, but I will do it, good point